src/struct.cpp

```diff
@@ -204,4 +209,14 @@ size_t StructManager::GetTuplesCnt() const
   return tuples_.size();
 }
 
+const Field *StructManager::GetProbeArg(const ast::Probe &probe,
+                                        const std::string &arg_name)
+{
+  auto args = Lookup(probe.args_typename());
+  if (args.expired() || !args.lock()->HasField(arg_name))
+    return nullptr;
+
+  return &args.lock()->GetField(arg_name);
```

Comment:
I think there's a potential race condition in this code (even if it won't be hit in the current implementation, if we're using shared_ptrs we should prepare for this as someone else might change the code in the future).

It'd be safer to do something like this instead:
```
if (auto observe = weak.lock()) {
  // use *observe
} else {
  return nullptr;
}
```
https://en.cppreference.com/w/cpp/memory/weak_ptr/lock